pingcap/tidb-binlog
# Add default value when generate the update rows

### What problem does this PR solve? 


### What is changed and how it works?
only on the condition PreWrite.schemaVersion < schemaVersion of the
table replicated to downstream.

### Check List 

Tests 


 - Manual test (add detailed scripts or steps below)

enable amend:
`
set global tidb_enable_amend_pessimistic_txn = 1;
`

run:
```
/* INIT */ drop table if exists t;
/* INIT */ create table t (id int primary key, c_str varchar(20));
/* INIT */ insert into t values (1, '0001'), (2, '0002'), (3, null), (4, '0003'), (5, null);
/* TXN */ begin;
/* TXN */ insert into t values (6, '0004');
/* TXN */ insert into t values (7, null);
/* DDL */ alter table t add c_str_new varchar(20);   # IN ANOTHER SESSION
/* TXN */ update t set c_str = '0005' where id = 1;
/* TXN */ update t set c_str = null where id = 2;
/* TXN */ update t set c_str = '0006' where id = 3;
/* TXN */ delete from t where id = 4;
/* TXN */ delete from t where id = 5;
/* TXN */ commit;
/* TXN */ select * from t order by id;
/* TXN */ admin check table t;

```

Code changes

 
Side effects



### Release note
- No Release note.
